File: core/src/main/java/by/itechart/common/dto/CreateUserDto.java
package by.itechart.common.dto;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class CreateUserDto {

    private String username;
    private String password;
    private String firstname;
    private String lastname;
    private String patronymic;
    private String email;
    private AddressDto address;
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    private LocalDate birth;
    private List<AuthorityDto> authorities;
}

File: web-app/online-warehouse/src/app/user/dto/create-user.dto.ts
import {AddressDto} from "../../shared/address/dto/address.dto";
import {AuthorityDto} from "./authority.dto";

export class CreateUserDto {
  username: string;
  password: string;
  firstname: string;
  lastname: string;
  patronymic: string;
  email: string;
  address: AddressDto;
  birth: Date;
  authorities: AuthorityDto[];
}
